lib/api/unittest/CAnnotationJsonWriterTest.cc

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#include <core/CJsonOutputStreamWrapper.h>
+
+#include <model/CAnnotation.h>
+#include <model/ModelTypes.h>
+
+#include <api/CAnnotationJsonWriter.h>
+
+#include <test/BoostTestCloseAbsolute.h>
+
+#include <rapidjson/document.h>
+#include <rapidjson/prettywriter.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <sstream>
+
+BOOST_AUTO_TEST_SUITE(CAnnotationJsonWriterTest)
+
+BOOST_AUTO_TEST_CASE(testWrite) {
+    std::ostringstream sstream;
+
+    {
+        ml::core::CJsonOutputStreamWrapper outputStream(sstream);
+        ml::api::CAnnotationJsonWriter writer(outputStream);
+
+        ml::model::CAnnotation annotation(1, "annotation text", 2, "pName", "pValue",
+                                          "oName", "oValue", "bName", "bValue");
+        writer.writeResult("job-id", annotation);
+    }
+
+    rapidjson::Document doc;
+    doc.Parse<rapidjson::kParseDefaultFlags>(sstream.str());
+    BOOST_TEST_REQUIRE(!doc.HasParseError());
+    const rapidjson::Value& firstElement = doc[0];
+    BOOST_TEST_REQUIRE(firstElement.HasMember("annotation"));
+    const rapidjson::Value& annotation = firstElement["annotation"];
+    BOOST_TEST_REQUIRE(annotation.HasMember("job_id"));
+    BOOST_REQUIRE_EQUAL(std::string("job-id"),
+                        std::string(annotation["job_id"].GetString()));
+    BOOST_TEST_REQUIRE(annotation.HasMember("timestamp"));
+    BOOST_REQUIRE_EQUAL(int64_t(1000), annotation["timestamp"].GetInt64());
```

Comment:
FYI it's not necessary any more to have the exact same types when asserting equality.  I know if you copy an existing file as an example you'll see this everywhere, but this is because CppUnit was really picky that both types in an equality assertion were the same.  Boost.Test does not mandate this.  So `int64_t(1000)` could simply be `1000` here.

I don't mind if you leave all these as-is for this PR, as we have the unnecessary casting and object construction all over the place in our tests due to the legacy of CppUnit.  But I just wanted to make sure you knew it's not necessary for the future.